packages/webcomponents/src/components/payment-details/get-payment-details.ts

```diff
@@ -0,0 +1,20 @@
+import { Payment } from '../../api';
+
+export const makeGetPaymentDetails =
+  ({ id, authToken, service }) =>
+  async ({ onSuccess, onError }) => {
+    try {
+      const response = await service.fetchPayment(id, authToken);
+      if (!response.error) {
+        onSuccess({ payment: new Payment(response.data) });
+      } else {
+        const responseError =
```

Comment:
Can we make a helper for this so that we don't have the possibility of handling response errors differently across components/services? Can we abstract even more of this so that the services are quick and easy to write, and consistent? It could be a separate follow-up PR after your pending PRs are closed

Reply:
Absolutely, that's a great idea! I've included the helper in the PR to ensure consistency across our components and services. Thanks for the suggestion!